client/state/jetpack-connect/reducer.js

```diff
@@ -26,6 +41,31 @@ export function jetpackConnectSite( state = {}, action ) {
 	return state;
 }
 
+export function jetpackConnectAuthorize( state = {}, action ) {
+	switch ( action.type ) {
+		case JETPACK_CONNECT_AUTHORIZE:
+			return Object.assign( {}, state, { isAuthorizing: true, authorizeSuccess: false, authorizeError: false } );
+		case JETPACK_CONNECT_AUTHORIZE_RECEIVE:
+			return Object.assign( {}, state, { isAuthorizing: false, authorizeError: action.error, authorizeSuccess: action.data, autoAuthorize: false } );
+		case JETPACK_CONNECT_QUERY_SET:
+			return Object.assign( {}, defaultAuthorizeState, { queryObject: action.queryObject } );
```

Comment:
I'm not 100% sure about this, but I believe that we need to 'clone' action.queryObject too here, because if we don't do it, { queryObject } is storing a pointer to the original action.queryObject ... I mean something like
```
const queryObject = Object.assign( {}, action.queryObject } );
return Object.assign( {}, defaultAuthorizeState, { queryObject: queryObject } );

```